walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,74 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
+/**
+ * Given a list of events, returns a list of possible time ranges for the meeting request.
+ */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Meeting duration cannot be longer than a day.
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+    if (events.isEmpty() || request.getAttendees().isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    
+    List<TimeRange> eventTimes = new ArrayList<>();
+    // Collect time ranges while filtering out non-attendees.
+    for (Event event : events) {
+      if (!Collections.disjoint(event.getAttendees(), request.getAttendees())) {
+        // The event and meeting request have at least one attendee in common.
+        eventTimes.add(event.getWhen());
+      }
+    }
+    if (eventTimes.isEmpty()) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    Collections.sort(eventTimes, TimeRange.ORDER_BY_START);
+
+    List<TimeRange> possibleTimes = new ArrayList<>();
+
+    // Check duration between start of day and start of first event.
```

Comment:
The comments are not clear what's the goal: Maybe something like "add start of the day if there is enough duration before first event". Focus on the output rather than the process.
Also, check usually means it's checking invariant, like state or input argument.
So does line 76